src/main/java/core/stack/MateStack.java

```diff
@@ -18,10 +20,16 @@ public void push(T value) {
     }
 
     public T peek() {
+        if (isEmpty()) {
+            throw new EmptyStackException();
+        }
         return stack[size - 1];
     }
 
     public T pop() {
+        if (isEmpty()) {
+            throw new EmptyStackException();
+        }
```

Comment:
You can create a separate method for this part. Because you call the same in another method.